airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -67,58 +72,87 @@ public JsonNode rowToJson(final ResultSet queryContext) throws SQLException {
     return jsonNode;
   }
 
+  @Override
+  protected void setDate(final PreparedStatement preparedStatement, final int parameterIndex, final String value) throws SQLException {
+    try {
+      Date date = Date.valueOf(value);
+      preparedStatement.setDate(parameterIndex, date);
+    } catch (final Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
   @Override
   public void setJsonField(final ResultSet resultSet, final int colIndex, final ObjectNode json) throws SQLException {
     final PgResultSetMetaData metadata = (PgResultSetMetaData) resultSet.getMetaData();
     final String columnName = metadata.getColumnName(colIndex);
-    final String columnTypeName = metadata.getColumnTypeName(colIndex);
+    final String columnTypeName = metadata.getColumnTypeName(colIndex).toLowerCase();
     final JDBCType columnType = safeGetJdbcType(metadata.getColumnType(colIndex));
-
     if (resultSet.getString(colIndex) == null) {
       json.putNull(columnName);
-    } else if (columnTypeName.equalsIgnoreCase("bool") || columnTypeName.equalsIgnoreCase("boolean")) {
-      putBoolean(json, columnName, resultSet, colIndex);
-    } else if (columnTypeName.equalsIgnoreCase("bytea")) {
-      putString(json, columnName, resultSet, colIndex);
-    } else if (columnTypeName.equalsIgnoreCase("time") || columnTypeName.equalsIgnoreCase("timetz")) {
-      putString(json, columnName, resultSet, colIndex);
     } else {
-      // https://www.postgresql.org/docs/14/datatype.html
-      switch (columnType) {
-        case BOOLEAN -> putBoolean(json, columnName, resultSet, colIndex);
-        case TINYINT, SMALLINT -> putShortInt(json, columnName, resultSet, colIndex);
-        case INTEGER -> putInteger(json, columnName, resultSet, colIndex);
-        case BIGINT -> putBigInt(json, columnName, resultSet, colIndex);
-        case FLOAT, DOUBLE -> putDouble(json, columnName, resultSet, colIndex);
-        case REAL -> putFloat(json, columnName, resultSet, colIndex);
-        case NUMERIC, DECIMAL -> putBigDecimal(json, columnName, resultSet, colIndex);
-        // BIT is a bit string in Postgres, e.g. '0100'
-        case BIT, CHAR, VARCHAR, LONGVARCHAR -> putString(json, columnName, resultSet, colIndex);
-        case DATE -> putDate(json, columnName, resultSet, colIndex);
-        case TIME -> putTime(json, columnName, resultSet, colIndex);
-        case TIMESTAMP -> putTimestamp(json, columnName, resultSet, colIndex);
-        case BLOB, BINARY, VARBINARY, LONGVARBINARY -> putBinary(json, columnName, resultSet, colIndex);
-        case ARRAY -> putArray(json, columnName, resultSet, colIndex);
-        default -> putDefault(json, columnName, resultSet, colIndex);
+      switch (columnTypeName) {
+        case "bool", "boolean" -> putBoolean(json, columnName, resultSet, colIndex);
```

Comment:
why do we need these extra `case` statements for `bool`, `boolean`, and `bytea` in the context of this PR?